src/I18n/Time.php

```diff
@@ -799,6 +838,100 @@ public static function parseTime($time, $format = null)
     }
 
     /**
+     * Returns the calendar type
+     * @param string|null $locale The locale name in which the date should be displayed (e.g. pt-BR)
+     * @return integer \IntlDateFormatter::TRADITIONAL|\IntlDateFormatter::GREGORIAN
+     */
+    public static function detectCalendarType($locale = null)
+    {
+        if (preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|coptic|ethiopic)/', $locale)) {
+            return \IntlDateFormatter::TRADITIONAL;
+        }
+        return \IntlDateFormatter::GREGORIAN;
+    }
+
+    /**
+     * Sets the default locale used when need when converting date to locale.
+     *
+     * @param string $calendar set calendar name.
+     * @return void
+     */
+    public static function setDefaultCalendar($calendar)
+    {
+        static::$defaultCalendar = $calendar;
+    }
+
+    /**
+     * Sets the default locale used when need when converting date to locale.
+     *
+     * @param string|null $locale The locale name in which the date should be displayed (e.g. pt-BR)
+     * @return void
+     */
+    public static function setDefaultLocale($locale)
+    {
+        static::$defaultLocale = $locale;
+    }
+
+    /**
+     * Returns current locale is setted.
+     *
+     * @param string|null $locale The locale name in which the date should be displayed (e.g. pt-BR)
+     * @param bool $useCalender can use default calendar name
+     * @return static|null
+     */
+    public static function getDefaultLocale($locale = null, $useCalender = false)
+    {
+        $locale = $locale ?: static::$defaultLocale;
+        
+        if ($locale === null || $locale === false) {
+            return null;
+        }
+
+        if (strpos($locale, '@') !== false ||
+            static::$defaultCalendar === null ||
+            $useCalender === false) {
+            return $locale;
+        }
+
+        if ($useCalender === false && static::$defaultCalendar === null) {
+            return $locale;
+        }
+
+        return $locale . '@calendar=' . static::$defaultCalendar;
+    }
+ 
+    /**
+     * Returns and convert DateTime pattern to Intl pattern
+     *
+     * @param string|int $pattern Any format accepted by DateTime pattern.
+     * @return string
+     */
+    public static function convertPattern($pattern = 'Y-m-d H:i:s')
+    {
+        if (!is_string($pattern) || $pattern === null) {
```

Comment:
wouldnt the first condition include the second? also, why returning the pattern then as it wouldnt match the expected return value, maybe return `''` instead?